File: src/Z.Web/System.Web.HttpResponse/HttpResponse.Redirect.cs
// Description: C# Extension Methods Library to enhances the .NET Framework by adding hundreds of new methods. It drastically increases developers productivity and code readability. Support C# and VB.NET
// Website & Documentation: https://github.com/zzzprojects/Z.ExtensionMethods
// Forum: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License: https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: http://www.zzzprojects.com/
// Copyright © ZZZ Projects Inc. 2014 - 2016. All rights reserved.
using System.Web;

public static partial class Extensions
{
    /// <summary>
    ///     A HttpResponse extension method that redirects.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <param name="urlFormat">The URL format.</param>
    /// <param name="values">A variable-length parameters list containing values.</param>
    public static void Redirect(this HttpResponse @this, string urlFormat, params object[] values)
    {
        string url = string.Format(urlFormat, values);
        @this.Redirect(url, true);
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Web/System.Web.HttpResponse/HttpResponse.Redirect.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Web

Public Module Extensions_858

	''' <summary>
	'''     A HttpResponse extension method that redirects.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <param name="urlFormat">The URL format.</param>
	''' <param name="values">A variable-length parameters list containing values.</param>
	<System.Runtime.CompilerServices.Extension> _
	Public Sub Redirect(this As HttpResponse, urlFormat As String, ParamArray values As Object())
		Dim url As String = String.Format(urlFormat, values)
		this.Redirect(url, True)
	End Sub
End Module


